docs/source/clickhouse.md

```diff
@@ -0,0 +1,134 @@
+│      	   0.1 │ ████▎                                                                        	      │
+│      	   0.2 │ █████████████▍                                                               	      │
+│      	   0.3 │ ████████████████████████                                                     	      │
+│      	   0.4 │ ████████████████████████████████████████████▋                                	      │
+│      	   0.5 │ ████████████████████████████████████████████████████████████████████▊        	      │
+│      	   0.6 │ ████████████████████████████████████████████████████████████████████████████████     │
+│      	   0.7 │ ██████████████████████████████████████████████████████████████████████       	      │
+│      	   0.8 │ ██████████████████████████████████████████                                   	      │
+│      	   0.9 │ ██████████▋                                                                  	      │
+│            1 │ ▌                                                                            	      │
+└──────────────┴──────────────────────────────────────────────────────────────────────────────────────┘
+```
+
+To get a deeper understanding about a dataset, ClickHouse provides statistical analysis functions for determining how your data is correlated, calculating statistical hypothesis tests, and more. Take a look at ClickHouse's [List of Aggregate Functions](https://clickhouse.com/docs/en/sql-reference/aggregate-functions/reference) for a complete list of available aggregate functions.
+
+## User-defined function (UDFs)
+
+A user-defined function (UDF) allows you to reuse custom logic. Many Hub datasets are often sharded into more than one Parquet file, so it can be easier and more efficient to create a UDF to list and query all the Parquet files of a given dataset from just the dataset name.
+
+For example, let's create a function to return a list of Parquet files from the [blog_authorship_corpus](https://huggingface.co/datasets/blog_authorship_corpus):
+
+```bash
+./clickhouse local -q "
+    CREATE OR REPLACE FUNCTION hugging_paths AS dataset -> (
+        SELECT arrayMap(x -> (x.1), JSONExtract(json, 'parquet_files', 'Array(Tuple(url String))'))
+        FROM url('https://datasets-server.huggingface.co/parquet?dataset=' || dataset, 'JSONAsString')
+    )
+
+    SELECT hf('blog_authorship_corpus') AS paths"
+
+['https://huggingface.co/datasets/blog_authorship_corpus/resolve/refs%2Fconvert%2Fparquet/blog_authorship_corpus/train/0000.parquet','https://huggingface.co/datasets/blog_authorship_corpus/resolve/refs%2Fconvert%2Fparquet/blog_authorship_corpus/train/0001.parquet','https://huggingface.co/datasets/blog_authorship_corpus/resolve/refs%2Fconvert%2Fparquet/blog_authorship_corpus/validation/0000.parquet']
+```
+
+Let's make this even easier by creating another function that calls `hugging_paths` and outputs all the files based on the dataset name:
+
+```bash
+./clickhouse local -q" 
```

Comment:
hugging_paths wont be available for use here - see above about need to start session OR specify hugging_paths function here as well